tree/ntupleutil/v7/src/RNTupleExporter.cxx

```diff
@@ -0,0 +1,124 @@
+/// \file RNTupleExporter.cxx
+/// \ingroup NTuple ROOT7
+/// \author Giacomo Parolini <giacomo.parolini@cern.ch>
+/// \date 2024-12-10
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <ROOT/RNTupleExporter.hxx>
+#include <ROOT/RPageStorage.hxx>
+#include <ROOT/RNTupleDescriptor.hxx>
+#include <ROOT/RClusterPool.hxx>
+#include <ROOT/RLogger.hxx>
+#include <fstream>
+#include <sstream>
+
+namespace ROOT::Experimental::Internal {
+
+struct RColumnExportInfo {
+   const RColumnDescriptor *fColDesc;
+   const RFieldDescriptor *fFieldDesc;
+   std::string fQualifiedName;
+
+   RColumnExportInfo(const RNTupleDescriptor &desc, const RColumnDescriptor &colDesc, const RFieldDescriptor &fieldDesc)
+      : fColDesc(&colDesc),
+        fFieldDesc(&fieldDesc),
+        fQualifiedName(desc.GetQualifiedFieldName(fieldDesc.GetId()) + '-' + std::to_string(colDesc.GetIndex()))
+   {
+   }
+};
+
+static void AddColumnsFromField(std::vector<RColumnExportInfo> &vec, const RNTupleDescriptor &desc,
```

Comment:
I believe this could move into an anonymous `namespace` to avoid it from being exported (even in `Internal`)